pyuvdata/mwa_corr_fits.py

```diff
@@ -69,9 +69,78 @@ def correct_cable_length(self, cable_lens):
+            The number of seconds to flag at the end of the observation. Set to
+            0 for no flagging.
+        flag_dc_offset: bool
+            Set to True to flag the center fine channel of each coarse channel.
+
+        Raises
+        ------
+        ValueError
+            If edge_width is not an integer multiple of the channel_width of the data (0 also acceptable).
+            If start_flag is not an integer multiple of the integration time (0 also acceptable).
+            If end_flag is not an integer multiple of the integration time (0 also acceptable).
+        """
+        if (edge_width % self.channel_width) > 0:
+            raise ValueError("The edge_width must be an integer multiple of the"
+                             "channel_width of the data or zero.")
+        if (start_flag % self.integration_time[0]) > 0:
+            raise ValueError("The start_flag must be an integer multiple of the"
+                             "integration_time of the data or zero.")
+        if (end_flag % self.integration_time[0]) > 0:
+            raise ValueError("The end_flag must be an integer multiple of the"
+                             "integration_time of the data or zero.")
+
+        num_ch_flag = int(edge_width / self.channel_width)
+        num_start_flag = int(start_flag / self.integration_time[0])
+        num_end_flag = int(end_flag / self.integration_time[0])
+
+        if num_ch_flag > 0:
+            edge_inds = []
+            for ch_count in range(num_ch_flag):
+                # count up from the left
+                left_chans = list(range(ch_count, self.Nfreqs, num_fine_chan))
+                # count down from the right
+                right_chans = list(range(self.Nfreqs - 1 - ch_count, 0, -num_fine_chan))
+                edge_inds = edge_inds + left_chans + right_chans
+
+            self.flag_array[:, :, edge_inds, :] = True
+
+        if flag_dc_offset:
+            center_inds = list(range(num_fine_chan // 2, self.Nfreqs, num_fine_chan))
+
+            self.flag_array[:, :, center_inds, :] = True
+
+        if (num_start_flag > 0) or (num_end_flag > 0):
+            shape = self.flag_array.shape
+            reshape = [self.Ntimes, self.Nbls, self.Nspws, self.Nfreqs, self.Npols]
+            self.flag_array = np.reshape(self.flag_array, reshape)
+            if num_start_flag > 0:
+                self.flag_array[:num_start_flag, :, :, :, :] = True
+            if num_end_flag > 0:
+                self.flag_array[-num_end_flag:, :, :, :, :] = True
+            self.flag_array = np.reshape(self.flag_array, shape)
+
     def read_mwa_corr_fits(self, filelist, use_cotter_flags=False, correct_cable_len=False,
                            phase_to_pointing_center=False, run_check=True, check_extra=True,
-                           run_check_acceptability=True):
+                           run_check_acceptability=True, flag_init=False,
+                           edge_width=80e3, start_flag=4.0, end_flag=6.0,
```

Comment:
Should this be start_flag=2.0, end_flag=2.0?

Reply:
Yes, thanks!